modules/presence/subscriber.c

```diff
@@ -280,6 +280,45 @@ static int presence_alloc(struct contact *contact)
 	return 0;
 }
 
+static void presence_dealloc(struct presence* pres)
+{
+	list_unlink(&pres->le);
+	tmr_cancel(&pres->tmr);
+	mem_deref(pres->contact);
+}
```

Comment:
You are right, the presence object must be deref'd as well.
But this is still necessary, since `pres->contact` is `mem_ref`'d in `presence_alloc`.